packages/js/product-editor/src/components/attribute-control/attribute-table-row.tsx

```diff
@@ -166,37 +225,70 @@ export const AttributeTableRow: React.FC< AttributeTableRowProps > = ( {
 		);
 	} );
 
-	async function addNewTerms(
-		termNames: string[],
-		selectedTerms: ProductAttributeTerm[]
-	) {
+	async function addNewTerms( newTerms: TokenItem[] ) {
 		if ( ! attribute ) {
 			return;
 		}
 
+		/*
+		 * Create the temporary terms.
+		 * It will show the tokens in the token field
+		 * optimistically, before the terms are created.
+		 */
+		setTemporaryTerms( ( prevTerms ) => [ ...prevTerms, ...newTerms ] );
+
 		// Create the new terms.
-		const promises = termNames.map( async ( termName ) => {
-			const newTerm = ( await createProductAttributeTerm( {
-				name: termName,
-				slug: cleanForSlug( termName ),
-				attribute_id: attributeId,
-			} ) ) as ProductAttributeTerm;
+		const promises = newTerms.map( async ( term ) => {
+			const newTerm = ( await createProductAttributeTerm(
+				{
+					name: term.value,
+					slug: cleanForSlug( term.value ),
+					attribute_id: attributeId,
+				},
+				{
+					optimisticQueryUpdate: {
+						search: '',
+						attribute_id: attributeId,
+					},
+					optimisticUrlParameters: [ attributeId ],
+				}
+			) ) as ProductAttributeTerm;
 
 			return newTerm;
 		} );
 
 		const newItems = await Promise.all( promises );
```

Comment:
Looks like an existing issue, but we aren't handling if any of the promises reject.

With the optimistic handling, this now results in an attribute added to the list that is always `validating` and can't be removed if the REST API to add the term fails.

Since this is an edge case, I'm okay with handling this in a follow up PR/issue if it isn't easy to address in this PR.